File: system-test/dotnet/SystemTests/Legacy/Mod08/E2eTests/PlaceOrderPositiveTest.cs
using Dsl.Core.UseCase;
using Driver.Port.Dtos;
using DomainValueTypes;
using SystemTests.Legacy.Mod08.E2eTests.Base;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Legacy.Mod08.E2eTests;

public class PlaceOrderPositiveTest : BaseE2eTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldPlaceOrderForValidInput(Channel channel)
    {
        await Scenario(channel)
            .Given().Product()
                .WithUnitPrice(20.00m)
            .When().PlaceOrder()
                .WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
                    .HasOrderNumberPrefix("ORD-")
                    .HasQuantity(5)
                    .HasUnitPrice(20.00m)
                    .HasStatus(OrderStatus.Placed)
                    .HasTotalPriceGreaterThanZero();
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod08/e2e/PlaceOrderPositiveTest.java
package com.mycompany.myshop.systemtest.legacy.mod08.e2e;

import com.mycompany.myshop.testkit.channel.ChannelType;
import com.mycompany.myshop.testkit.domainvaluetypes.OrderStatus;
import com.mycompany.myshop.systemtest.legacy.mod08.e2e.base.BaseE2eTest;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

class PlaceOrderPositiveTest extends BaseE2eTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldPlaceOrderForValidInput() {
        scenario
                .given().product().withUnitPrice(20.00)
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                .hasOrderNumberPrefix("ORD-")
                .hasQuantity(5)
                .hasUnitPrice(20.00)
                .hasStatus(OrderStatus.PLACED)
                .hasTotalPriceGreaterThanZero();
    }
}
